foundry/quadtree.d.ts

```diff
@@ -1 +1,152 @@
-declare class Quadtree {}
+declare interface QuadTreeObject<T> {
+  /**
+   * Rectangle of this object
+   */
+  r: Rectangle;
+  /**
+   * The stored data
+   */
+  t: T;
+  /**
+   * Set of quadtrees that hold this object
+   */
+  n: Set<Quadtree<T>>;
+}
+
+/**
+ * A Quadtree implementation that supports collision detection for rectangles.
+ */
+declare class Quadtree<T> {
+  /**
+   * @param bounds  - The outer bounds of the region
+   * @param options - Additional options which configure the Quadtree
+   */
+  constructor(bounds: Rectangle, { maxObjects, maxDepth, _depth }?: Quadtree.Options);
+
+  /**
+   * The bounding rectangle of the region
+   */
+  bounds: Rectangle;
+
+  /**
+   * The maximum number of objects allowed within this node before it must split
+   * @defaultValue 20
```

Comment:
One thing I only saw just now: We should make it distinct, whether the default value is just textually describing something or actually providing a code snippet. Code snippets should be either inline code or entire code blocks.
The same goes for all of the `@defaultValue` tags in this PR.
```suggestion
   * @defaultValue `20`
```